sdk/tests/utilities/test_id_generator.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+from . import IdGenerator
+from parameterized import parameterized
+
+
+class IdGeneratorTests(unittest.TestCase):
+
+    def test_use_default_scope_when_not_specified(self):
+        id_generator = IdGenerator()
+        self.assertEqual("sdk_example", id_generator.default_scope)
+
+    @parameterized.expand([
+        ["custom-scope"],
+        [""]
+    ])
+    def test_uses_provided_scope(self, scope):
+        id_generator = IdGenerator(scope)
+        self.assertEqual(scope, id_generator.default_scope)
```

Comment:
This test is (very) slightly confusing when you are testing for a `provided_scope` and asserting that against the `default_scope`, especially when the previous test is testing for the true `default_scope`.

Reply:
I see the challenge in that there are two kinds of default here. One is the class default and one is the class instance default.